File: backend/src/main/java/com/d4k/ecommerce/modules/recommendation/service/RecommendationService.java
package com.d4k.ecommerce.modules.recommendation.service;

import com.d4k.ecommerce.modules.product.dto.response.ProductResponse;
import java.util.List;

/**
 * Recommendation Service Interface
 * Cung cấp các phương thức đề xuất sản phẩm
 */
public interface RecommendationService {
    
    /**
     * Lấy sản phẩm tương tự (cùng category)
     * @param productId - ID sản phẩm đang xem
     * @param limit - Số lượng tối đa
     * @return Danh sách sản phẩm tương tự
     */
    List<ProductResponse> getSimilarProducts(Long productId, int limit);
    
    /**
     * Lấy sản phẩm phổ biến (được mua nhiều nhất)
     * @param limit - Số lượng tối đa
     * @return Danh sách sản phẩm phổ biến
     */
    List<ProductResponse> getPopularProducts(int limit);
}

File: frontend/src/services/recommendation-service.js
import apiClient from './api-client';

/**
 * Recommendation Service
 * API calls cho recommendation system
 */
const recommendationService = {
  /**
   * Lấy sản phẩm tương tự (cùng category)
   * @param {Number} productId - ID sản phẩm đang xem
   * @param {Number} limit - Số lượng tối đa (default: 8)
   * @returns {Promise}
   */
  getSimilarProducts: (productId, limit = 8) => {
    return apiClient.get(`/recommendations/similar/${productId}`, {
      params: { limit },
    });
  },

  /**
   * Lấy sản phẩm phổ biến (được mua nhiều nhất)
   * @param {Number} limit - Số lượng tối đa (default: 8)
   * @returns {Promise}
   */
  getPopularProducts: (limit = 8) => {
    return apiClient.get('/recommendations/popular', {
      params: { limit },
    });
  },
};

export default recommendationService;
